include/rocksdb/slice.h

```diff
@@ -222,7 +222,7 @@ class PinnableSlice : public Slice, public Cleanable {
  private:
   friend class PinnableSlice4Test;
   std::string self_space_;
-  std::string* buf_;
+  std::string* buf_ = nullptr;
```

Comment:
Should this instead be changing the PinnableSlice::operator around line 373 to set buf_=nullptr (or to something else)?  Under other constructors, buf_ is initialized okay.